resources/assets/js/components/DiscourseTopic.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+    <b-tr class="topic">
+        <b-td class="main-link"><a :href="url">{{ topic.title }}</a>
+            <div>
+                <a class="badge-wrapper" :href="categoryUrl">
+                    <span class="badge-category-bg"
+                          v-bind:style="{ backgroundColor: '#'+topic.category.color}"
+></span>
+                    <span data-drop-close="true" class="badge-category clear-badge" :title="topic.category.title">
+                        <span class="category-name">
+                            {{ topic.category.name }}
+                        </span>
+                    </span>
+                </a>
+            </div>
+        </b-td>
+        <b-td></b-td>
+        <b-td class="text-center">{{ topic.posts_count }}</b-td>
+        <b-td class="text-center">{{ ago }}</b-td>
+    </b-tr>
+</template>
+
+<script>
+import moment from 'moment'
+
+export default {
+  props: {
+    topic: {
+      type: Object,
+      required: true
+    },
+    discourseBaseUrl: {
+      type: String,
+      required: true
+    }
+  },
+  computed: {
+    ago() {
+      return new moment(this.topic.created_at).fromNow()
```

Comment:
Don't think it matters here too much - I don't think the topics are ever so recent that they need updating in terms of seconds or minutes.  But if it's simple to add in, why not (or at least to see how, for other places where the data is more regularly refreshed)

Reply:
What you do is start a timer to write the time to the store regularly:

```
    updateTime() {
      this.$store.dispatch('misc/setTime')
      this.timeTimer = setTimeout(this.updateTime, 10000)
    }
```

(You'll need to clear the timer too in beforeDestroy).

In the store you have the action and getter:

```
import Vue from 'vue'

export const state = () => ({
  cdn: process.env.CDN,
  time: null
})

export const mutations = {
  set(state, params) {
    Vue.set(state, params.key, params.value)
  },

  setTime(state) {
    state.time = new Date()
  }
}

export const getters = {
  get: state => key => state[key],
  time: state => state.time
}

export const actions = {
  set({ commit }, params) {
    commit('set', params)
  },

  setTime({ commit }) {
    commit('setTime')
  }
}

```

Then in a computed property you add a fake dependency on the time

```
    timeago() {
      let ret = null

      // Make depend on misc/time so that reactivity updates.
      if (this.$store.getters['misc/time'] && this.chatmessage) {
        ret = this.$dayjs(this.chatmessage.date).fromNow()

        // dayjs pluralises wrongly in some cases - we've seen 1 hours ago.
        ret = ret.replace(this.dePlural, '1 $1')
      }

      return ret
    }

```

So when the timer updates the store, reactivity forces a re-evaluation of the computed property.

The avoids the horror of having a zillion timers running, which can kill the browser.